datafusion/sql/src/select.rs

```diff
@@ -70,10 +71,37 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // process `where` clause
         let plan = self.plan_selection(select.selection, plan, planner_context)?;
 
+        // handle named windows before processing the projection expression
```

Comment:
I think the code would be nicer if this code to expand out named window definitions was refactored into a sub function -- something like `fn expand_window_definitions()` or something, perhaps